azure-pipelines.yml

```diff
@@ -12,8 +12,8 @@ jobs:
     name: macOS
     vmImage: macos-11
     matrix:
-      py37_np118:
-        PYTHON: '3.7'
+      py38_np118:
```

Comment:
I see that all the 3.7 tests were simply ported to 3.8. Is this OK, or do we want to take some of these tests to 3.9 or 3.10 or are these potentially slots for 3.11? Maybe it is also OK to merge this as is and "re-balance" the distribution of Python versions across tests in a later PR. @sklam @stuartarchibald what do you think?

Reply:
Running the function `numba.tests.support.print_azure_matrix()` will print the Azure CI matrix as a table to `stdout`. Suggest looking at the difference between `main` and this PR first?

Reply:
I manually counted up in this file where `main`
```
NumPy | Python | Count
-----------------------
 1.18 |  3.7   |   2
 1.18 |  3.8   |   1
 1.19.2=*_0 |  3.8   |   3
 1.19.2=*_0 |  3.9   |   1
 1.20 |  3.8   |   3
 1.20 |  3.9   |   1
 1.21 |  3.9   |   1
 1.21 |  3.10  |   1
 1.22 |  3.10  |   0
 1.22 |  3.8   |   1
 1.23 |  3.9   |   1
 1.23 |  3.10  |   2
```
and for this PR (all the 1.18/3.7 have been moved to 1.18/3.8):
```
NumPy | Python | Count
-----------------------
 1.18 |  3.8   |   3
 1.19.2=*_0 |  3.8   |   3
 1.19.2=*_0 |  3.9   |   1
 1.20 |  3.8   |   3
 1.20 |  3.9   |   1
 1.21 |  3.9   |   1
 1.21 |  3.10  |   1
 1.22 |  3.10  |   0
 1.22 |  3.8   |   1
 1.23 |  3.9   |   1
 1.23 |  3.10  |   2
```

Reply:
I think these changes are fine, thanks for checking.